src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1870,7 +1870,39 @@ protected override LocalState UnreachableState()
             return result;
         }
 
-#region Visitors
+        #region Visitors
+
+        public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
+        {
+            Debug.Assert(!IsConditionalState);
+
+            // Create slot when the state is unconditional since EnsureCapacity should be
+            // called on all fields and that is simpler if state is limited to this.State.
+            int slot = -1;
+            if (_performStaticNullChecks && this.State.Reachable)
+            {
+                var pattern = node.Pattern;
+                if (pattern.Kind == BoundKind.ConstantPattern && ((BoundConstantPattern)pattern).ConstantValue?.IsNull == true)
```

Comment:
The following pattern (which you handle) ensures x is null:
    x is null
The following patterns (which you do not handle) ensure x is not null
    x is Type y // where `Type` is not inferred via `var`
    x is K // where K is a constant other than null

Reply:
Thanks. I've added a `// PROTOTYPE` comment to handle those cases.

Reply:
I saw you had tests as well.